src/txrebroadcast.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <consensus/consensus.h>
+#include <logging.h>
+#include <miner.h>
+#include <script/script.h>
+#include <txrebroadcast.h>
+#include <util/time.h>
+#include <validation.h>
+
+#include <boost/multi_index/hashed_index.hpp>
+#include <boost/multi_index/member.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index_container.hpp>
+
+/** We rebroadcast 3/4 of max block weight to reduce noise due to circumstances
+ *  such as miners mining priority transactions. */
+static constexpr unsigned int MAX_REBROADCAST_WEIGHT{3 * MAX_BLOCK_WEIGHT / 4};
+
+/** Default minimum age for a transaction to be rebroadcast */
+static constexpr std::chrono::minutes REBROADCAST_MIN_TX_AGE{30min};
+
+/** Maximum number of times we will rebroadcast a transaction */
+static constexpr int MAX_REBROADCAST_COUNT{6};
+
+/** Minimum amount of time between returning the same transaction for
+ * rebroadcast */
+static constexpr std::chrono::hours MIN_REATTEMPT_INTERVAL{4h};
```

Comment:
this comment seems to have a lot of overlap with the one you left here: https://github.com/bitcoin/bitcoin/pull/21061#discussion_r615146994. I don't understand exactly what you mean by "cheap-RBF replacement". I'm going to resolve this conversation, but we can continue the topic in that other thread.

Reply:
See second half of other comment answer on why you can consider RBF cheap for an attacker. More generally there is also the fact that the bip125 replacement penalty is static (DEFAULT_INCREMENTAL_RELAY_FEE) is _static_ and doesn't scale up in function of your mempool congestion.

Simplify fee-bumping algorithms a lot though at the price of bandwidth overcost for network nodes processing replacement...

W.r.t to `filterInventoryKnown`, a malicious roll over isn't new from introducing node rebroadcast logic, though from now this module make assumptions on it to prevent bandwidth bleeding.